### PR TITLE
tpm2_getcap: fix segfault error for long property lists

### PR DESCRIPTION
**Pull request by cplappert**
ESAPI function Esys_GetCapability was always called with the same property
if the returned more_data flag was true. Addresses #1311.

Signed-off-by: cplappert <christian.plappert@sit.fraunhofer.de>

### CODE REVIEW

File: lib/tpm2_capability.c [hunk 1]
```diff
@@ -40,45 +40,102 @@
 
 #include "tpm2_capability.h"
 
-static size_t tpm2_get_property_data_size(TPM2_CAP capability) {
+static size_t tpm2_get_property_data_size(TPM2_CAP capability, size_t *maxSize) {
```
Comment (williamcroberts) [inline lib/tpm2_capability.c]: no camelCase... maxsize or max_size (preferred).

File: lib/tpm2_capability.c [hunk 2]
```diff
@@ -93,13 +150,18 @@ bool tpm2_capability_get (ESYS_CONTEXT *ectx,
 
     TPMI_YES_NO more_data;
     *capability_data = NULL;
+    size_t maxSize;
```
Comment (williamcroberts) [inline lib/tpm2_capability.c]:
```C
size_t max_size
```

File: lib/tpm2_capability.c [hunk 3]
```diff
@@ -93,13 +150,18 @@ bool tpm2_capability_get (ESYS_CONTEXT *ectx,
...
 
-    size_t property_size = tpm2_get_property_data_size(capability);
+    size_t property_size = tpm2_get_property_data_size(capability, &maxSize);
 
     if (!property_size) {
         return false;
     }
 
+    *capability_data = calloc(1, sizeof(TPMS_CAPABILITY_DATA));
```
Comment (williamcroberts) [inline lib/tpm2_capability.c]: this leaks

**Comment by cplappert** [comment]
Closing because already fixed by #1319

**Pull request closed without merging**